pds4indextools/pds4_create_xml_index.py

```diff
@@ -198,271 +701,781 @@ def update_nillable_elements_from_xsd_file(xsd_file, nillable_elements_info):
+        f".//*[local-name()='attribute' and @name='{target_name}']"
+        f"/descendant::*[local-name()='restriction']/@base",
+        f".//*[local-name()='simpleType' and @name='{target_name}']"
+        f"/*[local-name()='restriction']/@base",
+        f".//*[local-name()='simpleType' and @name='{target_name}']"
+        f"/descendant::*[local-name()='restriction']/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"//*[local-name()='extension']/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"//*[local-name()='extension']/*/*/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"//*[local-name()='extension']/*/*/*/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"//*[local-name()='extension']/*/*/*/*/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"//*[local-name()='extension']/*/@nilReason",
+        f".//*[local-name()='complexType' and @name='Science_Facets']"
+        f"//*[local-name()='element' and @name='{target_name}']/@type",
+        f".//xs:complexType[@name='{target_name}']"
+        f"//xs:extension[@base='pds:{target_name}_WO_Units']"
+        f"/xs:attribute[@name='unit']/@type",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"/descendant::*[local-name()='simpleContent']"
+        f"/*[local-name()='extension']/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"/descendant::*[local-name()='simpleContent']"
+        f"/*[local-name()='extension']/*/*/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"/descendant::*[local-name()='simpleContent']"
+        f"/*[local-name()='extension']/*/*/*/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"/descendant::*[local-name()='simpleContent']"
+        f"/*[local-name()='extension']/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"//*[local-name()='simpleContent']"
+        f"/*[local-name()='extension']/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"//*[local-name()='simpleContent']"
+        f"/*[local-name()='extension']/*/*/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"//*[local-name()='simpleContent']"
+        f"/*[local-name()='extension']/*/*/*/@base"
+    ]
+
+    for query in other_queries:
+        result = get_base_type(query)
+        if result:
+            base_type = result[0]
+            return follow_base_type(base_type)
+
+    return None
+
+
+def scrape_namespaces(xsd_url):
+    """
+    Fetch and parse an XSD file from a given URL to extract namespace declarations.
+
+    Parameters:
+    xsd_url (str): The URL of the XSD file to be fetched and parsed.
```

Comment:
It seems like a lot of these new docstrings don't have indents after the Parameters and Returns and Raises. They should.